code/modules/mob/living/carbon/human/species_types/dwarves.dm

```diff
@@ -54,21 +54,22 @@ GLOBAL_LIST_INIT(dwarf_last, world.file2list("strings/names/dwarf_last.txt")) //
 //Dwarf Speech handling - Basically a filter/forces them to say things. The IC helper
 /datum/species/dwarf/proc/handle_speech(datum/source, list/speech_args)
 	var/message = speech_args[SPEECH_MESSAGE]
-	if(message[1] != "*")
-		message = " [message]" //Credits to goonstation for the strings list.
-		var/list/dwarf_words = strings("dwarf_replacement.json", "dwarf") //thanks to regex too.
-
-		for(var/key in dwarf_words) //Theres like 1459 words or something man.
-			var/value = dwarf_words[key] //Thus they will always be in character.
-			if(islist(value)) //Whether they like it or not.
-				value = pick(value) //This could be drastically reduced if needed though.
-
-			message = replacetextEx(message, " [uppertext(key)]", " [uppertext(value)]")
-			message = replacetextEx(message, " [capitalize(key)]", " [capitalize(value)]")
-			message = replacetextEx(message, " [key]", " [value]") //Also its scottish.
-
-		if(prob(3))
-			message += pick(" By Armok!")
+	if(speech_args[SPEECH_LANGUAGE] != /datum/language/dwarf) // No accent if they speak their language
```

Comment:
Do an early return instead if the types match.

Reply:
Early returns would prevent the rest of the code of running, including parts outside of this `if`

Reply:
Oh, so the "By Armok!" bit at the end is intended regardless? Ok, nice.